common/etc/nginx/include/s3gateway.js

```diff
@@ -1022,6 +1031,39 @@ async function _fetchEC2RoleCredentials() {
     };
 }
 
+/**
+ * Get the credentials by assuming calling AssumeRoleWithWebIdentity with the environment variable
+ * values ROLE_ARN, AWS_WEB_IDENTITY_TOKEN_FILE and HOSTNAME
+ *
+ * @returns {Promise<{accessKeyId: (string), secretAccessKey: (string), sessionToken: (string), expiration: (string)}>}
+ * @private
+ */
+async function _fetchWebIdentityCredentials(r) {
+    var arn   = process.env['AWS_ROLE_ARN'];
+    var name  = process.env['HOSTNAME'] || 'nginx-s3-gateway';
+    var sts_endpoint = process.env['STS_ENDPOINT'] || 'https://sts.amazonaws.com';
```

Comment:
The [official documentation](https://docs.aws.amazon.com/eks/latest/userguide/configure-sts-endpoint.html) refers to an environment variable `AWS_STS_REGIONAL_ENDPOINTS` that can be either `regional` or `global`. This seems like an easier approach for configuring regional endpoints as it's controlled via an annotation on the service account. Not sure on this but some documentation might point to the annotation also controlling whether regional endpoints are available/required. [This section](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html#id_credentials_temp_enable-regions_writing_code) shows the format to obtain the regional endpoint.

Reply:
The documentation is relevant when you use the AWS SDK. As the SDK is not used at all, the environment variable will not have any effect.

Reply:
I filed #55 to add support for this.